### ISSUE TITLE
After upgrade to v9, project_key updates from default to null and fails

### ISSUE THREAD

**Issue by Nereis** [jfrog/terraform-provider-artifactory#822]
**Describe the bug**
Same issue as explain in #647. My guess would be it's related to #804 

Terraform Code
```
locals {
  remote_alpine = {
    r-alpine-dl-cdn-alpinelinux-org = { url = "https://dl-cdn.alpinelinux.org" },
  }
}

resource "artifactory_remote_alpine_repository" "standard_remote_alpine" {
  for_each = local.remote_alpine

  key                                   = each.key
  url                                   = each.value.url
  description                           = "Managed by Terraform"
  notes                                 = lookup(each.value, "notes", null)
  property_sets                         = ["artifactory"]
  retrieval_cache_period_seconds        = 600
  unused_artifacts_cleanup_period_hours = 1440
  xray_index                            = true
}
```
Terraform Plan
```
# module.arti-remotes.artifactory_remote_alpine_repository.standard_remote_alpine["r-alpine-dl-cdn-alpinelinux-org"] will be updated in-place
  ~ resource "artifactory_remote_alpine_repository" "standard_remote_alpine" {
        id                                    = "r-alpine-dl-cdn-alpinelinux-org"
      - project_key                           = "default" -> null
        # (30 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```
Terraform Apply
```
│ Error: 
│ 404 DELETE https://artifactory/access/api/v1/projects/_/attach/repositories/r-alpine-dl-cdn-alpinelinux-org
│ {
│   "errors" : [ {
│     "code" : "NOT_FOUND",
│     "message" : "Could not find resource `r-alpine-dl-cdn-alpinelinux-org`; type `repo` in Projects configuration"
│   } ]
│ }
│ 
│   with module.arti-remotes.artifactory_remote_alpine_repository.standard_remote_alpine["r-alpine-dl-cdn-alpinelinux-org"],
│   on ../../modules/terraform-arti-remotes/remote-alpine.tf line 7, in resource "artifactory_remote_alpine_repository" "standard_remote_alpine":
│    7: resource "artifactory_remote_alpine_repository" "standard_remote_alpine" {
```

Versions
- terraform: 1.5.6
- provider: 9.2.1
- artifactory: 7.55.8

**Expected behavior**
Resources created with an old version of the provider and using the default value should not fail on the new version of the provider. 

**Additional context**
Adding the project_key on all resources does workaround the issue
```
project_key  = "default"
```

**Comment by alexhung** [comment]
@Nereis As mentioned in the [changelog for V9.0.0 release](https://github.com/jfrog/terraform-provider-artifactory/blob/master/CHANGELOG.md#900-sep-15-2023-tested-on-artifactory-7687-with-terraform-cli-v157) this is to revert the Artifactory API behavior where `projectKey` contains `default` for repository that is not assigned to **any** project. The fix in Artifactory is released for v7.68.7 (self-hosted) and v7.67.0 (cloud). I noticed that you are still using v7.55.8 which means your instance does not yet have this fix. This correlates to your mention of the working by adding `project_key  = "default"`.

Per Semantic versioning, major version change contains breaking/not backward compatible changes. So my recommendation is to remain on v8.x until your Artifactory instance is upgraded to the latest (v7.68 or later).

**Comment by Nereis** [comment]
@alexhung  Well as soon as you upgraded, you can't rollback since the state won't let you do it and raise an exception about an older version used.
